crates/wasmtime/src/runtime.rs

```diff
@@ -730,8 +732,23 @@ pub struct Store {
 pub(crate) struct StoreInner {
     engine: Engine,
     compiler: RefCell<Compiler>,
+    interrupts: Arc<VMInterrupts>,
+    signatures: RefCell<SignatureRegistry>,
```

Comment:
IIRC `SignatureRegistry` has a lock internally, could that be removed now in favor of just using the `RefCell` here for mutability?

I think that there's no need for `SignatureRegistry` to be synchronized, even in a shared memory world I think instances are still disconnected in that they can have different signature registries on each thread.